dask/dataframe/io/csv.py

```diff
@@ -508,6 +510,15 @@ def read_pandas(
     else:
         path_converter = None
 
+    # If compression is "infer", inspect the (first) path suffix and
+    # set the proper compression option if the suffix is recongnized.
+    if compression == "infer":
+        # Translate the input urlpath to a simple path list
+        paths = get_fs_token_paths(urlpath, mode="rb", storage_options=kwargs)[2]
```

Comment:
Ran into an issue with `read_csv` on a remote source during dask-cudf s3 tests. (https://github.com/rapidsai/cudf/pull/7144)
Looks like storage options isn't being propagated here. 
`paths = get_fs_token_paths(urlpath, mode="rb", storage_options=storage_options)[2]`

Reply:
Thanks @ayushdg! cc @rjzamora

Reply:
Oops! I'll submit a quick fix. Sorry about that!